metaflow/plugins/events_decorator.py

```diff
@@ -598,8 +598,38 @@ def format_deploytime_value(self):
             old_trig = trigger
             if isinstance(trigger, DeployTimeField):
                 trigger = deploy_time_eval(trigger)
+                if is_stringish(trigger):
+                    pass
+                elif isinstance(trigger, dict):
+                    if "name" not in trigger:
+                        raise MetaflowException(
+                            "The *flow* attribute for *@trigger_on_finish* is missing the "
+                            "*name* key."
+                        )
+                    flow_name = trigger["name"]
+
+                    if not is_stringish(flow_name) or "." in flow_name:
+                        raise MetaflowException(
+                            "The *name* attribute of the *flow* is not a valid string"
+                        )
+                    result = {"fq_name": flow_name}
+                    if "project" in trigger:
+                        if is_stringish(trigger["project"]):
+                            result["project"] = trigger["project"]
+                        else:
+                            raise MetaflowException(
+                                "The *project* attribute of the *flow* is not a string"
+                            )
+                    if "project_branch" in trigger:
+                        if is_stringish(trigger["project_branch"]):
+                            result["branch"] = trigger["project_branch"]
+                        else:
+                            raise MetaflowException(
+                                "The *project_branch* attribute of the *flow* is not a string"
+                            )
+                    trigger = result
             if isinstance(trigger, dict):
```

Comment:
maybe add a comment on this code: "effect is to set all fields to None if they don't exist.